internal/common/autogen/handle_operations.go

```diff
@@ -154,70 +158,128 @@ func HandleDelete(ctx context.Context, req HandleDeleteReq) {
 		addError(d, opDelete, errCallingAPI, err)
 		return
 	}
-	// don't consider wait errors as it can happen that the resource is already deleted
-	_ = handleWait(ctx, req.Wait, req.Client, req.State)
+	if err := handleWaitDelete(ctx, req.Wait, req.Client); err != nil {
+		addError(d, opDelete, errWaitingForChanges, err)
+	}
 }
 
-// handleWait waits until a long-running operation is done if needed.
+// handleWaitCreateUpdate waits until a long-running operation is done if needed.
 // It also updates the model with the latest JSON response from the API.
-func handleWait(ctx context.Context, wait *WaitReq, client *config.MongoDBClient, model any) error {
+func handleWaitCreateUpdate(ctx context.Context, wait *WaitReq, client *config.MongoDBClient, model any) error {
 	if wait == nil {
 		return nil
 	}
-	respBody, err := waitForChanges(ctx, wait, client)
-	if err != nil {
+	bodyResp, err := waitForChanges(ctx, wait, client)
+	if err != nil || isEmptyJSON(bodyResp) {
 		return err
 	}
-	if respBody == nil {
+	if err := Unmarshal(bodyResp, model); err != nil {
```

Comment:
nit: Is there a need for this isEmptyJSON check? from refresh function implementation I understand empty json implies DELETED, so an error would be returned